base_layer/common_types/src/tari_address/dual_address.rs

```diff
@@ -583,4 +583,68 @@ mod test {
             payment_id.as_slice()
         );
     }
+
+    #[test]
+    fn valid_max_payment_id() {
+        // Generate random public key
+        let mut rng = rand::thread_rng();
+        let view_key = CompressedPublicKey::from_secret_key(&PrivateKey::random(&mut rng));
+        let spend_key = CompressedPublicKey::from_secret_key(&PrivateKey::random(&mut rng));
+        let payment_id = vec![1u8; MAX_ENCRYPTED_DATA_SIZE + 1];
```

Comment:
MAX_ENCRYPTED_DATA_SIZE has a different value in encrypted_data.rs, suggest adding `MAX_PAYLOAD_SIZE: usize = 256` to encrypted_data.rs and use it here
```suggestion
        let payment_id = vec![1u8; MAX_PAYLOAD_SIZE + 1];
```